runtime_scan/pkg/scanner/job_managment.go

```diff
@@ -86,16 +90,19 @@ func (s *Scanner) jobBatchManagement(scanDone chan struct{}) {
 		}(data, s.killSignal)
 	}
 
+	// wait for killSignal or fullScanDone
 	select {
 	case <-s.killSignal:
 		log.WithFields(s.logFields).Info("Scan process was canceled")
 	case <-fullScanDone:
 		log.WithFields(s.logFields).Infof("All jobs has finished")
 		// Nonblocking notification of a finished scan
+		// notify back to server on scan done
 		nonBlockingNotification(scanDone)
 	}
 }
 
+// worker waits for data on the queue, run a scan job and waits for results from that scan job. he then notify that he is done to the Scanner
```

Comment:
// worker waits for data on the queue, runs a scan job and waits for results from that scan job. Upon completion, done is notified to the caller